### ISSUE TITLE
Operator<< not parsed correctly in template class

### ISSUE THREAD

**Issue by AdrienVannson** [doxygen/doxygen#9852]
**Describe the bug**
Doxygen fails to parse operator<< on template class.

**To Reproduce**
Use the following code:
```c++
#include <iostream>
using namespace std;

template<class T>
class A;

template<class T>
std::ostream &operator<<(std::ostream &os, const A<T> &a)
{
    os << a.x;
    return os;
}

template<class T>
std::istream &operator>>(std::istream &is, A<T> &a)
{
    is >> a.x;
    return is;
}

template<class T>
class A {
    friend std::ostream &operator<< <T>(std::ostream &os, const A<T> &a);
    friend std::istream &operator>> <T>(std::istream &is, A<T> &a);

private:
    T x;
};

int main()
{
    A<int> a;
    cin >> a;
    cout << a << "\n";
}
```
Generate the documentation with:
```
doxygen -g
doxygen
```

On the documentation, `<<` is missing in `operator<<`.

![Screenshot](https://user-images.githubusercontent.com/11526087/218313034-0a8c087e-a2c1-47ca-bf23-742249d29745.png)

**Version**
I ran doxygen on Ubuntu 22.04.1 LTS. I tried with the default version (1.9.1) installed with APT and the version from Github (1.9.7 (752b64a47fb0676da0fe9f58e3a954d006ea6bc3)) that I compiled myself. The same problem occurs with both.

**Comment by doxygen** [comment]
Please verify if commit https://github.com/doxygen/doxygen/commit/99d4a28e28929af892cb116d60e4935f9c46bfbc fixes the problem.

**Comment by AdrienVannson** [comment]
It seems to fix the problem, thank you!

**Comment by albert-github** [comment]
Please don't close the issue as this will be done, automatically, at the moment of an official release.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
